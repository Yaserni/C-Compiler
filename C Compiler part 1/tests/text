function int chek (int x,y)
{
   var real f=5,x;
   if(f>5)
      x=x+1;

 return x;

}

function void main()
{
   var int z;
   z=check(5,z);
}
