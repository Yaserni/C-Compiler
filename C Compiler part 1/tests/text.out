(CODE
    (
        (FUNCTION chek
            (ARGS
                ( INT
                    ( x y)
                    
                )
            ) (TYPE INT)
            (BODY

                (VAR REAL
                    (= f 5)
                    
                )

                (IF
                    (> f 5)
                    
                    (= x
                        (+ x 1)
                        
                    )
                )
            )
            )
            (RETURN x)
            
        (FUNCTION MAIN (ARGS NONE) (TYPE VOID)
            (BODY

                (VAR INT
                    ( z)
                    
                )

                (FUNC_CALL z
                    ( 5
                        ( z)
                        
                    )
                )
            )
            )
        )
    )
)
Accepted
